docs/operators/transform.md

```diff
@@ -0,0 +1,137 @@
+Transform - Operator Documentation
+==================================
+
+About Transform operator
+------------------------
+
+Transform means mapping of field expression from input to output or conversion of fields from one type to another.
+This operator is stateless. This operator receives objects on its input port; for each such input object, it creates a new output object whose fields are computed as expressions involving fields of the input object. 
+The types of the input and output objects are configurable as are the expressions used to compute the output fields. 
+
+The operator class is TransformOperator located in the package com.datatorrent.lib.transform.
+
+Use Case
+--------
+
+Consider the data received from input source and want to transform the data as below output schema while emitting to downstream operator. 
+
+Consider input objects with these fields:
+
+`{FirstName, LastName, Phone, DateOfBirth, Address}`
+ 
+ and output objects with fields: 
+ 
+ `{Name, Phone, Age, Address}`. 
+ 
+Suppose Name is a concatenation of FirstName and LastName and 
+        Age is computed by subtracting the DateOfBirth from the current year.
+
+These simple computations can be expressed as Java expressions where the input object is
+represented by $ and provided as configuration parameters as follows:
+
+```
+Name => {$.FirstName}.concat(\" \").concat({$.LastName})
+Age => (new java.util.Date()).getYear() - {$.dateOfBirth}.getYear()
+```
+
+#### Configuration Parameters
+-----------------------------
+
+-   ***expressionMap*** -   Map<String, String>
+    -   Mandatory Parameter
+    -   Specifies the map between the output field and expression using fields of input Java object.
+
+-   ***expressionFunctions*** -   List<String>
+    -   List of imported classes or methods should be made available to expression to use. It overrides the default list.
+        Consider the same example which is explained under use case section and suppose DateOfBirth field is of type org.joda.time.DateTime.
+        org.joda.time.DateTime class is not present in the default list. So, we need to add this in expressionFunctions as below:
+        ```
+        TransformOperator operator = new TransformOperator();
+        operator.setExpressionFunctions(Arrays.asList("org.joda.time.DateTime", org.apache.commons.lang3.StringUtils.*));
+        ```
+        
+    -   Default Value = {java.lang.Math.*, org.apache.commons.lang3.StringUtils.*, org.apache.commons.lang3.StringEscapeUtils.*, org.apache.commons.lang3.time.DurationFormatUtils.*, org.apache.commons.lang3.time.DateFormatUtils.*}
+
+-   ***copyMatchingFields*** -   boolean
+    -   Specifies whether the name and type of the same fields between input and output should be copied as is or not.
```

Comment:
Suggest rewriting:
"Specifies whether matching fields should be copied; here _matching_ means the name and type of an input field is the same as the name and type of an output field."
Also, explain what happens when a matching field also appears in the expression map.